src/videojs-contrib-hls.js

```diff
@@ -134,27 +134,76 @@ export default class HlsHandler extends Component {
       this.tech_.audioTracks().addEventListener('change', this.audioTrackChange_);
     });
 
-    this.masterPlaylistController_.on('selectedinitialmedia', () => {
-      let audioTrackList = this.tech_.audioTracks();
-      let media = this.masterPlaylistController_.media();
-      let master = this.masterPlaylistController_.master();
-      let mediaGroups = master.mediaGroups;
-      let attributes = {
-        audio: {main: {default: true}}
-      };
-
-      if (!media.attributes) {
-        // source URL was playlist manifest, not master
-        // no audio tracks to add
+    this.masterPlaylistController_.on('audioinfochanged', (e) => {
+      if (!videojs.browser.IS_FIREFOX) {
         return;
       }
+      let audioTrackList = this.tech_.audioTracks();
+      let error = 'had different audio properties (channels, sample rate, etc.) ' +
+                  'or changed in some other way.  This behavior is currently ' +
+                  'unsupported in Firefox due to an issue: \n\n' +
+                  'https://bugzilla.mozilla.org/show_bug.cgi?id=1247138\n\n';
+
+      let attributes = this.masterPlaylistController_.getPlaylistAttributes();
+      let mainLabel;
+
+      for (let label in attributes.audio) {
+        let hlstrack = attributes.audio[label];
+
+        /* eslint-disable dot-notation */
+        // dot notation used on the key default to support ie8
```

Comment:
We do need IE8 to not throw an error when loading the script though. I can do some testing to verify that we need these. When we were working with webworkify they added an unquoted parameter called `default` that would error throw an error on IE8 while loading the script.